synapse/visibility.py

```diff
@@ -96,7 +96,7 @@ async def filter_events_for_client(
         if isinstance(ignored_users_dict, dict):
             ignore_list = frozenset(ignored_users_dict.keys())
 
-    erased_senders = await storage.main.are_users_erased(e.sender for e in events)
+    erased_senders = await storage.main.are_users_erased([e.sender for e in events])
```

Comment:
This was potential for a real bug.

Reply:
I don't think so? `are_users_erased` uses `simple_select_many_batch`, which does work correctly with generators, modulo not short-circuiting for an empty generator.

Reply:
Fair, it came out requiring that to be a collection.